packages/apputils/src/dialog.ts

```diff
@@ -281,23 +286,33 @@ class Dialog extends Widget {
   /**
    * Resolve a button item.
    */
-  private _resolve(item: Dialog.IButton): void {
+  private _resolve(button: Dialog.IButton): void {
     // Prevent loopback.
     let promise = this._promise;
     this._promise = null;
-    this.close();
     ArrayExt.removeFirstOf(Private.launchQueue, promise.promise);
-    promise.resolve(item);
+    let body = this._body;
+    let value: T | null = null;
+    if (body instanceof Widget && typeof body.getValue === 'function') {
+      value = body.getValue();
+    }
+    this.dispose();
+    promise.resolve({
+      button,
+      accept: button.accept,
```

Comment:
this seems a bit redundant. If accept always comes from the button, why include it as a separate property?